tests/test_lib_events.py

```diff
@@ -2358,3 +2360,105 @@ def test_12_set_max_failcount(self):
         self.assertEqual(fc, 123)
 
         remove_token("SPASS01")
+
+
+class CustomUserAttributesTestCase(MyTestCase):
+
+    def test_01_event_set_attributes_logged_in_user(self):
+
+        # Setup realm and user
+        self.setUp_user_realms()
+
+        g = FakeFlaskG()
+        audit_object = FakeAudit()
+        audit_object.audit_data["serial"] = "SPASS01"
+        g.logged_in_user = User("hans", self.realm1)
+        g.audit_object = audit_object
+
+        # The attributekey will be set as "test" and the attributevalue as "check"
+        options = {"g": g,
+                   "attrkey": "test",
+                   "attrvalue": "check",
+                   "handler_def": {
+                        "options": {"user": USER_TYPE.LOGGED_IN_USER}}
+                   }
+        t_handler = CustomUserAttributesHandler()
+        res = t_handler.do("set_custom_user_attributes", options=options)
+        self.assertTrue(res)
+
+        # Check that the user has the correct attribute
+        user = g.logged_in_user
+        a = user.attributes
+        self.assertEqual({'test': 'check'}, a)
+
+    def test_02_event_delete_attributes(self):
+
+        # Setup realm and user
+        self.setUp_user_realms()
+
+        g = FakeFlaskG()
+        audit_object = FakeAudit()
+        audit_object.audit_data["serial"] = "SPASS01"
+        g.logged_in_user = User("hans", self.realm1)
+        g.audit_object = audit_object
+        user = User("hans", self.realm1)
+        # Setup user attribute
+        ret = user.set_attribute('test', 'check')
+        self.assertTrue(ret)
+        a = user.attributes
+        self.assertEqual({'test': 'check'}, a)
+
+        # The eventhandler will delete the user-attribute
+        options = {"g": g,
+                   "attrkey": "test",
+                   "attrvalue": "check",
+                   "handler_def": {
+                       "options": {"user": USER_TYPE.LOGGED_IN_USER}}
+                   }
+        t_handler = CustomUserAttributesHandler()
+        res = t_handler.do("delete_custom_user_attributes", options)
+        self.assertTrue(res)
+
+        # Check that the user attribute is deleted
+        b = user.attributes
+        self.assertEqual({}, b)
+
+    def test_03_event_set_attributes_tokenowner(self):
+
+        # Setup realm and user
+        self.setUp_user_realms()
+
+        g = FakeFlaskG()
+        audit_object = FakeAudit()
+        audit_object.audit_data["serial"] = "SPASS01"
```

Comment:
i think we need to create a token and assign it to a user (which should not be the `logged_in_user`, we do not need a `logged_in_user` in this case). The user should then be passed as the `req.User`.